x-pack/plugins/security/public/management/management_service.ts

```diff
@@ -28,30 +28,40 @@ interface SetupParams {
   getStartServices: StartServicesAccessor<PluginStartDependencies>;
 }
 
-interface StartParams {
-  management: ManagementStart;
-}
-
 export class ManagementService {
   private license!: SecurityLicense;
   private licenseFeaturesSubscription?: Subscription;
+  private securitySection?: ManagementSection;
 
   setup({ getStartServices, management, authc, license, fatalErrors }: SetupParams) {
     this.license = license;
 
-    const securitySection = management.sections.getSection(ManagementSectionId.Security);
+    const sectionTitle = i18n.translate('xpack.security.management.section.title', {
+      defaultMessage: 'Section',
+    });
+
+    const sectionTip = i18n.translate('xpack.security.management.section.tip', {
+      defaultMessage: 'Control access to features and data',
+    });
+
+    this.securitySection = management.sections.register({
```

Comment:
@mattkime I'll leave it to you to decide, but I'm not seeing the benefit of moving this definition here from where it currently is. Most of the other sections (Ingest, Data, Alerts and Insights, and Stack) are X-Pack-only, so I wonder what makes this one special? I think it makes maintenance easier to have all of these sections defined in one place. I'm open to exceptions if there's a compelling reason though.

Reply:
This moved because it was easy to move. I agree that the others could move and perhaps they should - it would be a matter of determining where they should live and how it impacts dependencies. But I'd only want to move them if the end result was clear and obvious. 

I think its worthwhile to make a point that the management plugin doesn't need to own all the sections. If I get pushback I'll stand down - this isn't worth frustrating others over.

Reply:
Thanks for explaining your line of thought! I'm worried that this principle of "A seciton is defined by a plugin, as long as that plugin is the only one populating the section, otherwise it's defined in Management" wouldn't be obvious to someone coming to this code for the first time.

Imagine someone looking for the definition of the Security section because they want to surface a plugin inside of it. They'd need to understand that the security plugin exists (non-obvious, considering most of the other "apps" surfaced here are their own plugins), and dig into it to discover that it's registering its own section.

Once they've done this, what should they do next: add their plugin to the Security plugin? Add Security as a dependency? Or extract the registration of the Security section to Management?

It seems like this move introduces a lot of complexity to discovery and decision-making. And I'm still not sure I understand the benefit.

Personally, I think all sections should be registered in Management and not by plugins, because I think it improves discoverability. I introduced this change in https://github.com/elastic/kibana/pull/65796 and shared some more of my thoughts behind it there.

Reply:
@cjcenizal You've changed my mind, I'll put it back